workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java

```diff
@@ -34,30 +34,22 @@ public class WorkflowManagementService {
   public void deploy(SwadlView swadlView) {
     Workflow workflow = objectConverter.convert(swadlView.getSwadl(), Workflow.class);
     CamundaTranslatedWorkflowContext context = workflowEngine.translate(workflow);
+    checkIfNoPublishedVersionExists(workflow);
+    String deploy = null;
+    if (workflow.isToPublish()) {
+      deploy = workflowEngine.deploy(context);
+      setCurrentActiveVersionToInactive(workflow.getId());
+    }
+    VersionedWorkflow versionedWorkflow = toVersionedWorkflow(workflow, swadlView, deploy);
+    versionRepository.save(versionedWorkflow);
+  }
+
+  private void checkIfNoPublishedVersionExists(Workflow workflow) {
```

Comment:
The method name is a bit confusing for me. It throws an error if the workflow is not found in the versionRepository or if it is found there but the "published" field is set to False.  I would name it **checkPublishedVersionExists** rather. WDYT?

Reply:
the check is to make sure there is no workflow having the same id with 'published' field as false, if there is any, it throws an exception. We are not allowing to create a new version while there exists a non-published version.
'checkPublishedVersionExists' does not reflect this meaning.

Reply:
I changed again the name, hope it is a bit more clearer

Reply:
Cool thank you!